microsoft/azuredatastudio
# Export to Markdown

**Description**: Although I could have gotten completely lost in the weeds with refactoring the export functionality, I have kept this change as minimal as possible. In this PR, I've finished the wiring up for exporting results to Markdown files. The changes follow the same pattern as existing export formats. 

**Outstanding Work**:
* Need an icon for export to markdown (fortunately, markdown logo is public domain)
* Need localization - I have no idea how to add this, but will add it if there are any suggestions/pointers
* Requires #20647 (and once committed, this PR will be rebased onto main)
* Requires microsoft/sqlops-dataprotocolclient#68 and update of dataprotocolclient
* Requires microsoft/sqltoolsservice#1705 and update of sqltoolsservice version

**Testing**:
To test:
* Pull down microsoft/sqlops-dataprotocolclient#68
* `yarn compile` on sqlops-dataprotocolclient
* Pull down this PR
* In ADS, modify extensions/mssql/package.json to point to this the local PR build of dataprotocolclient
* `yarn compile` on ADS
* Pull down microsoft/sqltoolsservice#1705
* Build solution on sqltoolsservice
* Copy sqltoolsservice bin folder to extensions/mssql/sqltoolsservice drop
* Launch ADS
* Run a query that selects something
* Observe new button on grid for exporting to Markdown
* Observe new right click action to export to Markdown on the grid
* Try it! Observe that it outputs a markdown file
